src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheck.java

```diff
@@ -147,6 +152,29 @@
  *   &lt;property name=&quot;allowSamelineParameterizedAnnotation&quot; value=&quot;true&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>
+ * Example to allow only one and only parameterized annotation on the same line:
+ * </p>
+ * <pre>
+ * &#64;SuppressWarnings("deprecation") DataLoader loader; // ok
+ * &#64;Mock DataLoader loader; // violation
+ * </pre>
+ * <p>
+ * Use the following configuration to only validate annotations on methods:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;AnnotationLocation&quot;&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;METHOD_DEF&quot;/&gt;
+ *  &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example for above configuration to check only methods:
+ * </p>
+ * <pre>
+ * &#64;Override public int hashCode() { ... } // ok
```

Comment:
show problematic field with comment as "ok" to prove that it is ignored.

Reply:
Could not understand this! Please elaborate.

Reply:
please add example `@Deprecated @Mock private int field1; // ok`
to show that only methods are validated.

Reply:
```@NotNull @Mock DataLoader loader; //ok``` There is already same type of example.Please check.